MCprep_addon/MCprep_resources/optimize_scene.py

```diff
@@ -185,13 +199,13 @@ def execute(self, context):
                 FilterGlossy = 1
                 MaxSteps = 50
                 
-            if util.getCycles() == 2:
+            if util.get_cycles_version() == 2:
                 bpy.context.scene.render.tile_x = 32
                 bpy.context.scene.render.tile_y = 32
         
         elif CyclesComputeDeviceType == "CUDA" or CyclesComputeDeviceType == "HIP":
             if CurrentRenderDevice == "CPU":
-                if bpy.context.preferences.addons["cycles"].preferences.has_active_device():
+                if util.get_preferences("cycles").preferences.has_active_device():
```

Comment:
Careful - if you are indending to get the Cycles preferences setting, this function `get_preferences` only has an input which is a bpy context or none. I think what you want here is:

```
prefs = util.get_preferences(context)
cprefs = prefs.addons.get("cycles")
if cprefs is not None and cprefs.preferences.has_active_device():
    # do stuff
```

Consider grabbing cycles preference once towards the top of the function, and just referencing it where needed (noting that this usage of ".get" is safe, in case someone has cycles disabled which is possible).